### ISSUE TITLE
Assertion `idx <= m_v->size()' failed

### ISSUE THREAD

**Issue by bwlang** [kensung-lab/hypo#6]
Assuming you want questions like this here... please let me know if you'd prefer something else.

Could this be related to too many contigs?
Any other hint?

```
*****************************************************************************************************************************************************
Stage 1: 100%
Stage 2: 100%
1st stage: 16237.3s
2nd stage: 25299s
Total    : 41536.3s
Tmp size : 209385MB

Stats:
   No. of k-mers below min. threshold :   1391580984
   No. of k-mers above max. threshold :     25064298
   No. of unique k-mers               :   4879558594
   No. of unique counted k-mers       :   3462913312
   Total no. of k-mers                : 145066797137
   Total no. of reads                 :   1130840130
   Total no. of super-k-mers          :  31206344409
[Hypo::Utils] Info: Value of K chosen for the given genome size (2.4g): 17
[Hypo::Utils] Info: File size expected for the given genome size (2.4g) and cov (80): 384G
Given Command: hypo  -d ../releases/v02.1/ovi_v02.1.fasta -r @illumina_reads.fofn -b ../releases/v02.1/alignments/illumina/ovi_illumina_aligned.sorted.bam -c 80 -s 2.4g.
[Hypo::Utils] Info: Intermediate Files will NOT be stored.
[Hypo::Utils] Info: Beginning from stage: 0
RESOURCES ([SUK:KMC]: Running KMC done. ): TIME= 41537 sec; PEAK RSS (so far)= 2074MB; CURRENT RSS (so far)= 2074MB.
RESOURCES ([SUK:KMC]: Kmers Histogram done. ): TIME= 182.438 sec; PEAK RSS (so far)= 3131MB; CURRENT RSS (so far)= 3131MB.
[SolidKmers] Info: Error-threshold freq: 17, Lower-threshold freq: 17, Upper-threshold freq: 0, Mean-coverage: 46
RESOURCES ([SUK]: Finding cutoffs. ): TIME= 6.74e-06 sec; PEAK RSS (so far)= 3131MB; CURRENT RSS (so far)= 3131MB.
RESOURCES ([SUK]: Filling bitvectors. ): TIME= 93.1106 sec; PEAK RSS (so far)= 3131MB; CURRENT RSS (so far)= 3131MB.
RESOURCES ([SUK]: Clearing files. ): TIME= 0.0311168 sec; PEAK RSS (so far)= 3131MB; CURRENT RSS (so far)= 3131MB.
[SolidKmers] Info: Number of solid kmers found: 0
RESOURCES ([SolidKmers]: Overall. ): TIME= 41820.3 sec; PEAK RSS (so far)= 3643MB; CURRENT RSS (so far)= 3131MB.
RESOURCES ([Hypo:Hypo]: Computed Solid kmers. ): TIME= 41821.9 sec; PEAK RSS (so far)= 3643MB; CURRENT RSS (so far)= 2074MB.
[Hypo::Hypo] Info: Number of (canonical) solid kmers (nonhp) : 0
RESOURCES ([Hypo:Hypo]: Loaded Contigs. ): TIME= 59.0052 sec; PEAK RSS (so far)= 3939MB; CURRENT RSS (so far)= 3939MB.
RESOURCES ([Hypo:Hypo]: Found Solid pos in contigs. ): TIME= 191.67 sec; PEAK RSS (so far)= 4015MB; CURRENT RSS (so far)= 4015MB.
********** [Hypo::Hypo] Info: BATCH-ID: 0
Aln processed (in 100 M): 0
[Hypo::Hypo] Info: Number of alignments (Batch 0): loaded (16983842) invalid (33884)
RESOURCES ([Hypo:Hypo]: Loaded alignments. ): TIME= 76.0928 sec; PEAK RSS (so far)= 9691MB; CURRENT RSS (so far)= 9691MB.
Kmer support update: 0
hypo: /opt/conda/conda-bld/hypo_1578830836910/work/external/suk/external/sdsl-lite/include/sdsl/rank_support_v.hpp:116: sdsl::rank_support::size_type sdsl::rank_support_v<b, t_patter_len>::rank(sdsl::rank_support::size_type) const [with unsigned char t_b = 1; unsigned char t_pat_len = 1; sdsl::rank_support::size_type = long unsigned int]: Assertion `idx <= m_v->size()' failed.
run_hypo.sh: line 1: 20067 Aborted                 hypo -d ../releases/v02.1/ovi_v02.1.fasta -r @illumina_reads.fofn -b ../releases/v02.1/alignments/illumina/ovi_illumina_aligned.sorted.bam -c 80 -s 2.4g
```

**Comment by Ritu-Kundu** [comment]
Hi Brad,

The number of contigs should not be a problem. This problem is originating in sdsl data-structure and needs further debugging. Could you please send us the files.

**Comment by bwlang** [comment]
I can but it's quite a lot of data...  
300G of of illumina reads + 250G bam (not sure why hypo needs both...)
47G of ONT bam
2.4G of draft
Do you have a location where I can put this?

**Comment by Ritu-Kundu** [comment]
Hi Brad,

Please run the same command with an additional flag `-i`.
It will generate 2 files in a folder name `aux` in your working directory: `stage.txt` and `solid_kmers.bvsd`
Please upload those 2 files along with the bam and  draft files.
Where should we send the instructions to upload the files? Your email-id on GitHub?

**Comment by bwlang** [comment]
Hi @Ritu-Kundu:

I can share those files if you like, but I found an inconsistency in my input that might explain this. Probably good to decent and print a user-actionable error message.
I provided a bam file containing RNA-seq reads instead of the bam file that I intended to use (which matches up with the fastq file).

Let me know if you'd like me to upload those files.
Feel free to use the email in my profile.
